packages/ripple-binary-codec/src/types/st-object.ts

```diff
@@ -109,11 +110,21 @@ class STObject extends SerializedType {
 
     const xAddressDecoded = Object.entries(value).reduce((acc, [key, val]) => {
       let handled: JsonObject | undefined = undefined
+      let updatedVal: JSON = val
       if (val && isValidXAddress(val.toString())) {
         handled = handleXAddress(key, val.toString())
         checkForDuplicateTags(handled, value)
+      } else if (key === PERMISSION_VALUE) {
```

Comment:
I think this is the wrong place to do this. It should be done in the same place as `TransactionType`/`LedgerEntryType` processing.

Reply:
I _think_ that's done here: https://github.com/XRPLF/xrpl.js/blob/main/packages/ripple-binary-codec/src/enums/xrpl-definitions-base.ts#L94-L97